### DISCUSSION TITLE
How to create a player group (with players of different brands/ecosystems) ?

### DISCUSSION THREAD

**Discussion by marcelveldt** [music-assistant/support#441] [Q&A]
You can stream the same music to multiple devices (of different brands), for example a mix of Sonos and Cast devices etc. by using player groups, created with the "helper" feature of Home Assistant.

Music Assistant will stream the same music stream to all players within the group.

**How to create a player group ?**

- Within Home Assistant, go to Settings --> Integrations
- Click on the "helpers" Tab
- Click the button at the button to create a new helper.
- Choose "Group" as helper type
- Choose "Media player group" as group type
- Add the entities you'd like to group
- Finish the wizard
- From within the Home Assistant Settings, navigate to Integrations
- Click "Configure" on the Music Assistant card
- Add the newly created group to the list of media players managed by Music Assistant
- Finish the Msic Assistant config wizard and allow it some time to reload with the new settings
- Voila, note that you now have a grouped player in the Music Assistant interface and you can stream music to it, regardless the type of players in that group.

**Some notes**

- When you start playback to a group, the stream will only be sent to players that are powered on at that moment
- When the last client of a group powers down, the group is also powered down (and playback stopped)
- The music will not be synchronized (playing at the exact same time) - This is however something that will be added soon for supported players

**Comment by Mariusthvdb** [comment]
having a lot of logic for excluding unavailable players and creating dynamic groups for the available ones in my config currently, this seems a great feature.

I take it the Music Assistant creates a `media_player.mass_xyz`, based on the helper group player `media_player.xyz`?

As such, can we use that `media_player.mass_xyz` also outside of Music Assistant and never see the errors of Google Chrome not able to find a player etc etc, when sending a TTs message/music stream/whatever to that mass group outside of Music Assistant?

**Reply by marcelveldt** [reply]
Yes that is the expectation. There may be some small bugs in there I did not account for but basically this is how it works. And it will even improve as soon as supported players can be synced.

**Comment by DutchJaFO** [comment]
possible issue for Sonos speakers :

1. create a speaker group within HA containing all sonos speakers
2. also group those speakers inside Sonos App
3. use the Music Assistant tab in HA to start playback of an album and target the HA group from step 1
4. expected result : music starts
5. actual result : playback fails with a stream encoding related error
6. ungroup those speakers inside the Sonos App
7. try step 3 again ...
8. playback starts

Speakers in the group are a mix of Symfonysk (Ikea), Play:1 (the original) and a Port (second generation)

**Reply by marcelveldt** [reply]
That is a strange bug and very good diagnose of you to find the root cause. I only have one Sonos test speaker here so no grouping possible but is it even possible with Sonos to stream to a single speaker when they are grouped ?

For example in the google cast ecosystem this is possible but maybe Sonos is different.

**Reply by DutchJaFO** [reply]
I suspect it is because it tries to stream to all grouped speakers and MA disrupts that attempt by also streaming to each separately?

possible situation :

Two speakers named 'A' and 'B'

MA streams to 'A' and then tries to stream to 'B' as well
Except Sonos itself is also trying to stream from 'A' to 'B' as well, which creates a conflict of sorts.

Possible solution :
= ungroup the grouped speakers before starting the stream and restore the original group at the end of the stream ?
OR
= ignore the grouped Sonos speakers and only stream to one of them in the group ?
This might cause issues if the HA group does not match the Sonos Group ...

Problem : what if someone changes the sonos group after the stream started ?

**Comment by OzGav** [comment]
This is outdated v1 info
